episodes/preparing_for_machine_learning.md

```diff
@@ -0,0 +1,460 @@
+
+Hint: Carefully examine the shape of the cardiomegaly image. Consider the impact of harsh lines on ML performance.
+
+:::::::::::::::::::::::: solution 
+
+## Solution
+ 
+```python
+# figure out how much to cut on sides
+print("cut top/bottom:", (image_cxr_cmegaly.shape[0] - image_cxr1.shape[0])/2)
+cut_top_bottom = abs(round((image_cxr_cmegaly.shape[0] - image_cxr1.shape[0])/2))
+# figure our how much to cut on top and bottom
+print("cut sides:",(image_cxr_cmegaly.shape[1] - image_cxr1.shape[1])/2)
+
+```
+
+```output
+cut top/bottom: -119.0
+cut sides: -208.5
+```
+
+
+```python
+cut_sides = abs(round((image_cxr_cmegaly.shape[1] - image_cxr1.shape[1])/2))
+list_images = [image_cxr1, image_cxr_cmegaly, image_cxr2]
+better_for_ml_list = []
+for image in list_images:
+    image = rotate(image,2)
+    image = image[cut_top_bottom:-cut_top_bottom, cut_sides: -cut_sides]
+    better_for_ml_list.append(image)
+
+# create figure for display
+fig = plt.figure(figsize=(10, 7))
+  
+# setting values to rows and column variables
+rows = 1
+columns = 3
+
+# add a subplot at the 1st position
+fig.add_subplot(rows, columns, 1)
+# showing image
+plt.imshow(better_for_ml_list[0])
+plt.axis('off')
+plt.title("Normal 1")
+  
+# add a subplot at the 2nd position
+fig.add_subplot(rows, columns, 2)
+# showing image
+plt.imshow(better_for_ml_list[1])
+plt.axis('off')
+plt.title("Augment 1")
+
+# add a subplot at the 3nd position
+fig.add_subplot(rows, columns, 3)
+# showing image
+plt.imshow(better_for_ml_list[2])
+plt.axis('off')
+plt.title("Augment 2")
+
+
+```
+
+![](fig/augmented_cxr_rotate.png){alt='augmented chest x-ray'}
+
+There are a few key considerations to keep in mind regarding the choices made here. The goal is to create data that realistically represents what might occur in real-life scenarios. For example, a 2-degree rotation is far more realistic than a 25-degree rotation. Unless you are using a rotationally invariant algorithm or another advanced method that accounts for such differences, even small rotations can make every pixel slightly different for the computer. A 2-degree rotation closely mirrors what might happen in a clinical setting, making it a more practical choice.
+
+Additionally, the results can be further improved by cropping the images. Generally, we aim to minimize harsh, artificial lines in any images used for ML algorithms, unless those lines are consistent across all images. It is a common observation among practitioners that many ML algorithms tend to "focus" on these harsh lines, so if they are not uniformly present, it is better to remove them. As for how much to crop, we based our decision on the size of our images. While you don't need to follow our exact approach, it is important to ensure that all images are ultimately the same size and that all critical features are retained.
```

Comment:
- "[...] it is important to ensure that all images are ultimately the same size", I agree, but "Augment 1" image (`better_for_ml_list[2]`) is bigger than "Normal 1" and "Augment 2". I think they should be the same size, right?
- Remove the "1" from "Normal 1" title

Reply:
Resized...I think the 1 is sort of an idea to keep track, but it could change with every augment. I'll do it that.